### ISSUE TITLE
Command crashes

### ISSUE THREAD

**Issue by bakkbone** [Hunter19823/kubejsoffline#7]
After running `/kubejsoffline`, the thread fails, logging the following:

```
[23:16:57] [KJSOffline DocThread/INFO]: Starting Documentation Thread...
[23:16:57] [KJSOffline DocThread/ERROR]: Uncaught exception in thread "KJSOffline DocThread"
java.lang.NoClassDefFoundError: net/minecraft/class_310
	at pie.ilikepiefoo.kubejsoffline.DocumentationThread.sendMessage(DocumentationThread.java:136) ~[kubejsoffline-3.1.6.jar:?]
	at pie.ilikepiefoo.kubejsoffline.DocumentationThread.run(DocumentationThread.java:51) ~[kubejsoffline-3.1.6.jar:?]
Caused by: java.lang.ClassNotFoundException: net.minecraft.class_310
	at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:525) ~[?:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:226) ~[fabric-loader-0.15.7.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:119) ~[fabric-loader-0.15.7.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:525) ~[?:?]
	... 2 more
```

1.20.1 Fabric, have tried with both v4.0.0 and v3.1.6 of the mod.

Mod list:
```
	- architectury 9.2.14
	- audioplayer 1.20.1-1.9.0
	   \-- admiral 0.4.4+1.20.1
	- bettersafebed 1.20-9
	- bookshelf 20.1.10
	- cardinal-components 5.2.2
	   |-- cardinal-components-base 5.2.2
	   |-- cardinal-components-block 5.2.2
	   |-- cardinal-components-chunk 5.2.2
	   |-- cardinal-components-entity 5.2.2
	   |-- cardinal-components-item 5.2.2
	   |-- cardinal-components-level 5.2.2
	   |-- cardinal-components-scoreboard 5.2.2
	   \-- cardinal-components-world 5.2.2
	- catalogue 1.8.0
	- cloth-config 11.1.118
	   \-- cloth-basic-math 0.6.1
	- configured 2.2.3
	   |-- com_electronwill_night-config_core 3.6.6
	   \-- com_electronwill_night-config_toml 3.6.6
	- coordinatesdisplay 7.0.0
	   \-- boxlib 9.2.0
	- coroutil 1.20.1-1.3.7
	- dailydad_server 1.4.0
	- discord-mc-chat 2.3.2
	   |-- com_fasterxml_jackson_core_jackson-annotations 2.14.1
	   |-- com_fasterxml_jackson_core_jackson-core 2.14.1
	   |-- com_fasterxml_jackson_core_jackson-databind 2.14.1
	   |-- com_google_code_gson_gson 2.10.1
	   |-- com_neovisionaries_nv-websocket-client 2.14
	   |-- com_squareup_okhttp3_okhttp 4.11.0
	   |-- com_squareup_okio_okio 2.10.0
	   |-- discord-mc-chat-1_20_1 2.3.2
	   |-- net_dv8tion_jda 5.0.0-beta.18
	   |-- net_fellbaum_jemoji 1.3.3
	   |-- net_sf_trove4j_trove4j 3.0.3
	   |-- org_apache_commons_commons-collections4 4.4
	   |-- org_jetbrains_kotlin_kotlin-stdlib 1.6.20
	   \-- org_slf4j_slf4j-api 2.0.9
	- easyanvils 8.0.2
	- enhancedgroups 1.20.1-1.3.0
	- essential_commands 0.35.0-mc1.20.1
	   |-- ec-core 1.3.0-mc1.20.1
	   |    |-- fabric-permissions-api-v0 0.2-SNAPSHOT
	   |    |-- placeholder-api 2.1.1+1.20
	   |    \-- playerabilitylib 1.8.0
	   |-- fabric-permissions-api-v0 0.2-SNAPSHOT
	   |-- org_yaml_snakeyaml 2.0
	   |-- placeholder-api 2.1.1+1.20
	   \-- playerabilitylib 1.8.0
	- fabric-api 0.92.1+1.20.1
	   |-- fabric-api-base 0.4.31+1802ada577
	   |-- fabric-api-lookup-api-v1 1.6.36+1802ada577
	   |-- fabric-biome-api-v1 13.0.13+1802ada577
	   |-- fabric-block-api-v1 1.0.11+1802ada577
	   |-- fabric-block-view-api-v2 1.0.1+1802ada577
	   |-- fabric-blockrenderlayer-v1 1.1.41+1802ada577
	   |-- fabric-client-tags-api-v1 1.1.2+1802ada577
	   |-- fabric-command-api-v1 1.2.34+f71b366f77
	   |-- fabric-command-api-v2 2.2.13+1802ada577
	   |-- fabric-commands-v0 0.2.51+df3654b377
	   |-- fabric-containers-v0 0.1.64+df3654b377
	   |-- fabric-content-registries-v0 4.0.11+1802ada577
	   |-- fabric-convention-tags-v1 1.5.5+1802ada577
	   |-- fabric-crash-report-info-v1 0.2.19+1802ada577
	   |-- fabric-data-attachment-api-v1 1.0.0+de0fd6d177
	   |-- fabric-data-generation-api-v1 12.3.4+1802ada577
	   |-- fabric-dimensions-v1 2.1.54+1802ada577
	   |-- fabric-entity-events-v1 1.6.0+1c78457f77
	   |-- fabric-events-interaction-v0 0.6.2+1802ada577
	   |-- fabric-events-lifecycle-v0 0.2.63+df3654b377
	   |-- fabric-game-rule-api-v1 1.0.40+1802ada577
	   |-- fabric-item-api-v1 2.1.28+1802ada577
	   |-- fabric-item-group-api-v1 4.0.12+1802ada577
	   |-- fabric-key-binding-api-v1 1.0.37+1802ada577
	   |-- fabric-keybindings-v0 0.2.35+df3654b377
	   |-- fabric-lifecycle-events-v1 2.2.22+1802ada577
	   |-- fabric-loot-api-v2 1.2.1+1802ada577
	   |-- fabric-loot-tables-v1 1.1.45+9e7660c677
	   |-- fabric-message-api-v1 5.1.9+1802ada577
	   |-- fabric-mining-level-api-v1 2.1.50+1802ada577
	   |-- fabric-model-loading-api-v1 1.0.3+1802ada577
	   |-- fabric-models-v0 0.4.2+9386d8a777
	   |-- fabric-networking-api-v1 1.3.11+1802ada577
	   |-- fabric-networking-v0 0.3.51+df3654b377
	   |-- fabric-object-builder-api-v1 11.1.3+1802ada577
	   |-- fabric-particles-v1 1.1.2+1802ada577
	   |-- fabric-recipe-api-v1 1.0.21+1802ada577
	   |-- fabric-registry-sync-v0 2.3.3+1802ada577
	   |-- fabric-renderer-api-v1 3.2.1+1802ada577
	   |-- fabric-renderer-indigo 1.5.1+1802ada577
	   |-- fabric-renderer-registries-v1 3.2.46+df3654b377
	   |-- fabric-rendering-data-attachment-v1 0.3.37+92a0d36777
	   |-- fabric-rendering-fluids-v1 3.0.28+1802ada577
	   |-- fabric-rendering-v0 1.1.49+df3654b377
	   |-- fabric-rendering-v1 3.0.8+1802ada577
	   |-- fabric-resource-conditions-api-v1 2.3.8+1802ada577
	   |-- fabric-resource-loader-v0 0.11.10+1802ada577
	   |-- fabric-screen-api-v1 2.0.8+1802ada577
	   |-- fabric-screen-handler-api-v1 1.3.30+1802ada577
	   |-- fabric-sound-api-v1 1.0.13+1802ada577
	   |-- fabric-transfer-api-v1 3.3.5+8dd72ea377
	   \-- fabric-transitive-access-wideners-v1 4.3.1+1802ada577
	- fabricloader 0.15.7
	   \-- mixinextras 0.3.5
	- fallingtree 4.3.4
	- forgeconfigapiport 8.0.0
	- framework 0.6.27
	   |-- org_javassist_javassist 3.29.2-GA
	   \-- org_reflections_reflections 0.10.2
	- ftblibrary 2001.2.0
	- ftbultimine 2001.1.4
	- jade 11.8.0
	- jadeaddons 5.2.3
	- java 17
	- jei 15.3.0.4
	- kubejs 2001.6.5-build.7
	- kubejsoffline 4.0.0
	- leashmod 1.0.6
	- leavesbegone 8.0.0
	- lithium 0.11.2
	- lootr 0.7.30.77
	- luckperms 5.4.102
	   \-- fabric-permissions-api-v0 0.2-SNAPSHOT
	- mcwfences 1.1.1
	- midnightlib 1.4.1
	- minecraft 1.20.1
	- modernfix 5.15.0+mc1.20.1
	- mr_vanilla_refresh 1.4.19h
	- oth3r-sit 1.1.3+1.20-1.20.1
	- pickupnotifier 8.0.0
	- puzzleslib 8.1.18
	   \-- puzzlesaccessapi 8.0.7
	- refurbished_furniture 1.0.0-beta.3
	- rhino 2001.2.2-build.18
	- server-hats 0.16
	- strawstatues 8.0.3
	   \-- puzzlesapi 8.1.4
	        \-- puzzlesaccessapi 8.0.7
	- tinyskeletons 8.0.1
	- tradingpost 8.0.2
	- travelersbackpack fabric-1.20.1-9.1.12
	   \-- reach-entity-attributes 2.4.0
	- trinkets 3.7.2
	- universal_shops 1.3.2+1.20.1
	   |-- common-economy-api 1.1.1
	   |-- common-protection-api 1.0.0
	   |-- fabric-permissions-api-v0 0.2-SNAPSHOT
	   |-- placeholder-api 2.1.1+1.20
	   |-- polymer-core 0.5.3+1.20.1
	   |    |-- polymer-networking 0.5.3+1.20.1
	   |    |    \-- polymer-common 0.5.3+1.20.1
	   |    |         \-- packet_tweaker 0.4.0+1.19.4
	   |    \-- polymer-registry-sync-manipulator 0.5.3+1.20.1
	   |-- polymer-resource-pack 0.5.3+1.20.1
	   |    \-- polymer-common 0.5.3+1.20.1
	   |         \-- packet_tweaker 0.4.0+1.19.4
	   |-- polymer-virtual-entity 0.5.3+1.20.1
	   |    \-- polymer-common 0.5.3+1.20.1
	   |         \-- packet_tweaker 0.4.0+1.19.4
	   |-- server_translations_api 2.0.0+1.20
	   |    \-- packet_tweaker 0.4.0+1.19.4
	   \-- sgui 1.2.2+1.20
	- villagerinabucket 1.20.1-1.1.1
	- villagernames 4.5.2
	- visualworkbench 8.0.0
	- voicechat 1.20.1-2.5.12
	   \-- fabric-key-binding-api-v1 1.0.36+fb8d95da82
	- watut 1.20.1-1.1.1
	- xaerominimap 24.1.1
	- xaeroworldmap 1.38.4
	- yet_another_config_lib_v3 3.2.2+1.20
	   |-- com_twelvemonkeys_common_common-image 3.10.0-SNAPSHOT
	   |-- com_twelvemonkeys_common_common-io 3.10.0-SNAPSHOT
	   |-- com_twelvemonkeys_common_common-lang 3.10.0-SNAPSHOT
	   |-- com_twelvemonkeys_imageio_imageio-core 3.10.0-SNAPSHOT
	   |-- com_twelvemonkeys_imageio_imageio-metadata 3.10.0-SNAPSHOT
	   |-- com_twelvemonkeys_imageio_imageio-webp 3.10.0-SNAPSHOT
	   |-- org_quiltmc_parsers_gson 0.2.1
	   \-- org_quiltmc_parsers_json 0.2.1
```

I'm unsure whether this is a mod conflict or a bug?

**Comment by Hunter19823** [comment]
Hmm this seems to be a classloading issue, possibly caused by my recent update, I'll have to research this further

**Comment by Hunter19823** [comment]
I did not run into this issue on the latest fabric version, I am now testing with the version 0.15.7 that you are using to see if that is the issue.

**Comment by Hunter19823** [comment]
It seems the class that isn't being loaded in your crash report is `net.minecraft.class_310`, which remaps to `net/minecraft/client/Minecraft`. Which is quite odd. Is it possible you are running this command on a dedicated server perhaps?

**Comment by Hunter19823** [comment]
> It seems the class that isn't being loaded in your crash report is `net.minecraft.class_310`, which remaps to `net/minecraft/client/Minecraft`. Which is quite odd. Is it possible you are running this command on a dedicated server perhaps?

This indeed is the problem. It's my mistake to never test this on a server.
I should be able to patch this issue relatively easily.

**Comment by Hunter19823** [comment]
This is fixed in the 1.19.2 build.
I am now working on the 1.20.1 build

**Comment by Hunter19823** [comment]
The patch is now on 1.20.1, however, something happened when publishing the build causing Curseforge to automatically mark all the versions for manual review, so I have to wait till that is resolved for it to be released.
